src/libcore/intrinsics.rs

```diff
@@ -520,6 +507,35 @@ extern "rust-intrinsic" {
     pub fn bswap32(x: u32) -> u32;
     /// Reverses the bytes in a `u64`.
     pub fn bswap64(x: u64) -> u64;
+}
+/// The following intrinsics are disallowed on PNaCl:
+/// We sneakily redirect them to libm or to a larger sized intrinsic.
```

Comment:
Not directly, but the PNaCl/JS IR simplification passes don't rewrite all those intrinsics to their `libm` counterparts (I assume because doing so would force a `libm` dependency on all pexe's). So we have to do it here (and in `librustc_trans/trans/context.rs`, but I haven't packaged that into a PR yet).

In fact most of the intrinsics are actually mapped to `libm` in `librustc_trans`. These functions are defined here because they could be made to work with casts and/or don't have direct `libm` counterparts.